src/courses/mappers/04.md

```diff
@@ -0,0 +1,342 @@
+---
+order: 4
+next: 05.md
+title: What Is an OHDF Mapper?
+author: Charles Hu
+---
+
+## What Is a Mapper?
+
+A mapper is a design pattern used to correlate (or map) items in two different objects with one another. Mappers are useful in that they allow us to connect items in objects that are nominally different but semantically the same. A result of this is that we can easily transform one object type into another through the use of mappings which define the direct relationship of items in each object type.
+
+Here are some scenarios which demonstrate some key aspects of mappers:
+
+::: details Transferring IDs
+Say we have to transfer the credentials of an employee who is moving from Business A to Business B. The data formats the businesses use for IDs are as follows:
```

Comment:
why would company a facilitate an employee moving to company b lol.  i like the actual example but i think the story here is weak.  a better one might be that the company is changing the software they are using to keep track of their employees and needs to move all of their data over into the new software.

Reply:
Addressed.